src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java

```diff
@@ -730,6 +1005,37 @@ public String toString()
         {
             return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
         }
+
+        @Override
+        public BoundKind kind()
+        {
+            return BoundKind.MULTI_CELL_UDT;
+        }
+
+        private static final BoundSerializer<MultiCellUdtBound> serializer = new BoundSerializer<MultiCellUdtBound>()
+        {
+            @Override
+            public void serialize(MultiCellUdtBound bound, DataOutputPlus out, int version) throws IOException
+            {
+                serializeList(bound.values, out, version, vintNullableSerializer);
+                out.writeUnsignedVInt(bound.protocolVersion.asInt());
+            }
+
+            @Override
+            public MultiCellUdtBound deserialize(DataInputPlus in, int version, ColumnMetadata column, Operator operator) throws IOException
+            {
+                List<ByteBuffer> values = deserializeList(in, version, vintNullableSerializer);
+                int protocolVersion = checkedCast(in.readUnsignedVInt());
+                // Does decode actually do what we want?
```

Comment:
It might not, given what it looks like I did in the `TxnCondition.Value` serializer (which itself was unnecessary, given where [CASSANDRA-18114](https://issues.apache.org/jira/browse/CASSANDRA-18114) is going).

Reply:
Protocol version is null for user type. I could remove this and pass null and it would continue to work and might be slightly clearer.